src/hidimstat/base_variable_importance.py

```diff
@@ -197,3 +204,157 @@ def plot_importance(
+            else:
+                self._feature_groups_ids = [
+                    np.array(ids, dtype=int)
+                    for ids in list(self.feature_groups.values())
+                ]
+        else:
+            raise ValueError("feature_groups needs to be a dictionary")
+        if isinstance(self.feature_types, str):
+            if self.feature_types in ["auto", "continuous", "categorical"]:
+                self.feature_types = [
+                    self.feature_types for _ in range(self.n_feature_groups_)
+                ]
+            else:
+                raise ValueError(
+                    "feature_types support only the string 'auto', 'continuous', 'categorical'"
+                )
+        return self
+
+    def _check_fit(self, X):
+        """
+        Check if the perturbation method has been properly fitted.
+
+        This method verifies that the perturbation method has been fitted by checking
+        if required attributes are set and if the number of features matches
+        the feature grouped variables.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Input data to validate against the fitted model.
+
+        Raises
+        ------
+        ValueError
+            If the method has not been fitted (i.e., if n_feature_groups_,
+            or _feature_groups_ids attributes are missing).
+        AssertionError
+            If the number of features in X does not match the total number
+            of features in the grouped variables.
+        """
+        if self.n_feature_groups_ is None or self._feature_groups_ids is None:
+            raise ValueError("The class is not fitted.")
+        if isinstance(X, pd.DataFrame):
+            names = list(X.columns)
+        elif isinstance(X, np.ndarray) and X.dtype.names is not None:
+            names = X.dtype.names
+            # transform Structured Array in pandas array for a better manipulation
+            X = pd.DataFrame(X)
+        elif isinstance(X, np.ndarray):
+            names = None
+        else:
+            raise ValueError("X should be a pandas dataframe or a numpy array.")
+        number_columns = X.shape[1]
+        for index_variables in self.feature_groups.values():
+            if isinstance(index_variables[0], numbers.Integral):
+                assert np.all(
+                    np.array(index_variables, dtype=int) < number_columns
+                ), "X does not correspond to the fitting data."
+            elif type(index_variables[0]) is str or np.issubdtype(
+                type(index_variables[0]), str
+            ):
+                assert np.all(
+                    [name in names for name in index_variables]
+                ), f"The array is missing at least one of the following columns {index_variables}."
+            else:
+                raise InternalError(
+                    "A problem with indexing has happened during the fit."
+                )
```

Comment:
Does this belong to _check_fit? it seems more like checking the input data rather than the status of the class instance.